odxtools/nameditemlist.py

```diff
@@ -69,10 +66,19 @@ def append(self, item: T):
             else:
                 tmp = f"{item_name}_{i}"
 
-    def sort(self, key=None, reverse=False):
+    def sort(self, key: Optional[Callable[[T], str]] = None, reverse: bool = False):
         return self._list.sort(key=key, reverse=reverse)
 
-    def __contains__(self, x) -> bool:
+    def keys(self) -> Iterable[str]:
+        return self.__dict__.keys()
+
+    def values(self) -> Iterable[T]:
+        return self.__dict__.values()
+
+    def items(self) -> Iterable[Tuple[str, T]]:
```

Comment:
* You re-invented the dict, subclass `collections.UserDict` instead
* Those changes are not related to original PR, please move them to their own PR

Reply:
Good point, but this class is also (primarily?) a list, so it also needs to inherit from `UserList`. I thus tried

```python
class NamedItemList(UserDict[str, T], UserList[T]):
  [...]
```

But this produced mypy errors a la
```
odxtools/nameditemlist.py:11: error: Definition of "data" in base class "UserDict" is incompatible with definition in base class "UserList"  [misc]
```

any idea?

Reply:
Could you move those changes to their own PR? we can discuss there possible solutions

Reply:
done. I figured that the patch can be removed completely. the `ProgCodeSyntax` patch now gets its separate PR...